packages/fiori/src/SideNavigation.ts

```diff
@@ -0,0 +1,387 @@
+import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
+import customElement from "@ui5/webcomponents-base/dist/decorators/customElement.js";
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+import ResponsivePopover from "@ui5/webcomponents/dist/ResponsivePopover.js";
+import fastNavigation from "@ui5/webcomponents-base/dist/decorators/fastNavigation.js";
+import event from "@ui5/webcomponents-base/dist/decorators/event.js";
+import property from "@ui5/webcomponents-base/dist/decorators/property.js";
+import slot from "@ui5/webcomponents-base/dist/decorators/slot.js";
+import { getI18nBundle } from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import type I18nBundle from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import List from "@ui5/webcomponents/dist/List.js";
+import StandardListItem from "@ui5/webcomponents/dist/StandardListItem.js";
+import Tree from "@ui5/webcomponents/dist/Tree.js";
+import TreeItem from "@ui5/webcomponents/dist/TreeItem.js";
+import type { TreeItemClickEventDetail } from "@ui5/webcomponents/dist/Tree.js";
+import SideNavigationItem from "./SideNavigationItem.js";
+import SideNavigationSubItem from "./SideNavigationSubItem.js";
+import SideNavigationTemplate from "./generated/templates/SideNavigationTemplate.lit.js";
+import SideNavigationPopoverTemplate from "./generated/templates/SideNavigationPopoverTemplate.lit.js";
+import {
+	SIDE_NAVIGATION_POPOVER_HIDDEN_TEXT,
+	SIDE_NAVIGATION_COLLAPSED_LIST_ARIA_ROLE_DESC,
+	SIDE_NAVIGATION_COLLAPSED_LIST_ITEMS_ARIA_ROLE_DESC,
+	SIDE_NAVIGATION_LIST_ARIA_ROLE_DESC,
+	SIDE_NAVIGATION_LIST_ITEMS_ARIA_ROLE_DESC,
+} from "./generated/i18n/i18n-defaults.js";
+
+// Styles
+import SideNavigationCss from "./generated/themes/SideNavigation.css.js";
+import SideNavigationPopoverCss from "./generated/themes/SideNavigationPopover.css.js";
+
+type SideNavigationPopoverContents = {
+	mainItem: SideNavigationItem,
+	mainItemSelected: boolean,
+	selectedSubItemIndex: number,
+	subItems: Array<SideNavigationSubItem>,
+};
+
+type TSideNavigationItem = SideNavigationItem | SideNavigationSubItem;
```

Comment:
I only would prefer to have ISideNavigationItem interface (instead of TSideNavigationItem), that both SideNavigationItem and SideNavigationSubItem implement

Reply:
We discussed this comment offline and it won't help much to have an interface in this case.